plugins/controlboard/src/ControlBoardDriver.cpp

```diff
@@ -1144,13 +1159,120 @@ bool GazeboYarpControlBoardDriver::setPIDsForGroup_IMPEDANCE(std::vector<std::st
+    {
+        yError()<<"Missing WRAPPER group";
+        return false;
+    }
+
+    if (!m_pluginParameters.findGroup("WRAPPER").check("networks"))
+    {
+        yError()<<"Missing networks group";
+        return false;
+    }
 
     yarp::os::Bottle& name_bot = m_pluginParameters.findGroup("WRAPPER").findGroup("networks");
     std::string name = name_bot.get(1).toString();
 
-    yarp::os::Bottle& kin_chain_bot = m_pluginParameters.findGroup(name);
+    motorControlGroup_bot = m_pluginParameters.findGroup(name);
+    return true;
+}
+
+bool GazeboYarpControlBoardDriver::setPositionsToleranceLinear()
+{
+    yarp::os::Bottle kin_chain_bot;
+    if (!FindMotorControlGroup(kin_chain_bot))
+        return false;
+
+    if (!kin_chain_bot.check("positionToleranceLinear")) {
+        yWarning()<<"No positionToleranceLinear value found in ini file, default one will be used!";
+        yDebug()<<"positionToleranceLinear: [ "<<m_robotPositionToleranceLinear<<" ]";
+        return true;
+    }
+
+    yInfo()<<"positionToleranceLinear param found!";
+    yarp::os::Bottle& positionToleranceLinear_bot = kin_chain_bot.findGroup("positionToleranceLinear");
+    if (static_cast<size_t>(positionToleranceLinear_bot.size()) != 2) {
+        yError()<<"Invalid number of params:positionToleranceLinear:"<<static_cast<size_t>(positionToleranceLinear_bot.size());
+        return false;
+    }
+
+    auto tmp=positionToleranceLinear_bot.get(1);
+    if (!tmp.isDouble())
+    {
+        yError()<<"Invalid param type:positionToleranceLinear";
+        return false;
+    }
+
+    m_robotPositionToleranceLinear=tmp.asDouble();       
+
+    yDebug()<<"positionToleranceLinear: [ "<<m_robotPositionToleranceLinear<<" ]";
+
+    return true;
+}
+
+bool GazeboYarpControlBoardDriver::setPositionsToleranceRevolute()
+{
+    yarp::os::Bottle kin_chain_bot;
+    if (!FindMotorControlGroup(kin_chain_bot))
+        return false;
+
+    if (!kin_chain_bot.check("positionToleranceRevolute")) {
+        yWarning()<<"No positionToleranceRevolute value found in ini file, default one will be used!";
+        yDebug()<<"positionToleranceRevolute: [ "<<m_robotPositionToleranceRevolute<<" ]";
+        return true;
+    }
+
+    yInfo()<<"positionToleranceRevolute param found!";
+    yarp::os::Bottle& positionToleranceRevolute_bot = kin_chain_bot.findGroup("positionToleranceRevolute");
+    if (static_cast<size_t>(positionToleranceRevolute_bot.size()) != 2) {
+        yError()<<"Invalid number of params:positionToleranceRevolute:"<<static_cast<size_t>(positionToleranceRevolute_bot.size());
+        return false;
+    }
+
+    auto tmp=positionToleranceRevolute_bot.get(1);
+    if (!tmp.isDouble())
+    {
+        yError()<<"Invalid param type:positionToleranceRevolute";
+        return false;
+    }
+
+    m_robotPositionToleranceRevolute=tmp.asDouble();       
+
+    yDebug()<<"positionToleranceRevolute: [ "<<m_robotPositionToleranceRevolute<<" ]";
```

Comment:
I think we can remove this print.

Reply:
I prefer to keep, also because is present also in the other methods

Reply:
I see, ref https://github.com/robotology/gazebo-yarp-plugins/pull/334#discussion_r160383752 .